inferno-core/src/Inferno/Infer.hs

```diff
@@ -1158,10 +1158,16 @@ unifies err (TSeries t1) (TSeries t2) = unifies err t1 t2
 unifies err (TOptional t1) (TOptional t2) = unifies err t1 t2
 unifies err (TTuple ts1) (TTuple ts2)
   | length (tListToList ts1) == length (tListToList ts2) = unifyMany err (tListToList ts1) (tListToList ts2)
+  | List.null err =
+      let pos = initialPos ""
+          loc = (pos, pos)
+       in throwError [UnificationFail Set.empty (TTuple ts1) (TTuple ts2) loc]
   | otherwise = throwError [UnificationFail (getTypeClassFromErrs err) (TTuple ts1) (TTuple ts2) loc | loc <- (getLocFromErrs err)]
```

Comment:
Looking at this code, the error throwing doesn't really make sense. My original design is definitely wonky here. The idea behind throwing a list was to link several places in code which generate a contradiction. e.g. somewhere we assign a variable the value bool and elsewhere we try to add a number to it. the error should point to those two places in the code as a contradiction. However, I'm not sure this ever properly worked as intended. It seems we always just throw a singleton error anyway.

To get proper error location, we should probably track location for types we are unifying and then add those to the error when throwing, instead of adding a dummy `pos`. This will require a bit of a re-engineering tho.